src/main/java/kr/co/jeelee/demoboard/domain/category/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package kr.co.jeelee.demoboard.domain.category.service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import kr.co.jeelee.demoboard.domain.category.dao.CategoryRepository;
+import kr.co.jeelee.demoboard.domain.category.dto.response.CategoryResponse;
+import kr.co.jeelee.demoboard.global.exception.custom.CustomException;
+import kr.co.jeelee.demoboard.global.exception.custom.ErrorCode;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+	private final CategoryRepository categoryRepository;
+
+	@Override
+	public List<CategoryResponse> findAll(final boolean isPostCount) {
+		Map<Long, Long> countPostByCategoryId;
+
+		if (isPostCount) {
+			countPostByCategoryId = categoryRepository.countPostByCategoryId().stream()
+				.collect(Collectors.toMap(
+					result -> (Long) result[0],
+					result -> (Long) result[1]
+				));
+		} else {
+			countPostByCategoryId = null;
+		}
+
+		return categoryRepository.findAll().stream()
+			.map(categoryEntity -> {
+				Long postCount = isPostCount
+					? countPostByCategoryId.get(categoryEntity.getId())
+					: null;
+
+				return CategoryResponse.of(categoryEntity, postCount);
```

Comment:
위에서 countPostByCategoryId 가 isPostCount 에 의해서 이미 판단이 한번 됐기 때문에 뒤에서 또 삼항 연잔사 쓸 필요 없을 것 같아요.
어차피 false 면 null 일꺼라서.. 한줄로 가능할꺼 같아요!
```suggestion
				return CategoryResponse.of(categoryEntity, countPostByCategoryId.get(categoryEntity.getId()));
```